libs/vkd3d/device.c

```diff
@@ -6563,13 +6563,33 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateSharedHandle(d3d12_device_if
     const struct vkd3d_vk_device_procs *vk_procs;
     struct DxvkSharedTextureMetadata metadata;
     ID3D12Resource *resource_iface;
+    OBJECT_ATTRIBUTES attr = {0};
     ID3D12Fence *fence_iface;
+    UNICODE_STRING name_str;
+    WCHAR buffer[MAX_PATH];
 
     vk_procs = &device->vk_procs;
 
     TRACE("iface %p, object %p, attributes %p, access %#x, name %s, handle %p\n",
             iface, object, attributes, access, debugstr_w(name), handle);
 
+    attr.Length = sizeof(attr);
+    attr.SecurityDescriptor = (void *)attributes;
+    if (name)
+    {
+        DWORD session, len, name_len = wcslen(name);
+
+        ProcessIdToSessionId(GetCurrentProcessId(), &session);
+        len = swprintf(buffer, ARRAY_SIZE(buffer), L"\\Sessions\\%u\\BaseNamedObjects\\", session);
```

Comment:
WCHAR is uint16_t on Linux build, but L"" is 32-bit on native. I assume all of this Win32 stuff can be ifdeffed out on Linux anyway?

Reply:
Right, then this is already in an ifdef _WIN32 afaics?

Reply:
We override WCHAR to uint16_t on Linux and use u"" to get forced 16-bit wchar since we have to consume that.